docs/docs/using-pants/advanced-target-selection.mdx

```diff
@@ -38,6 +38,10 @@ By default, `--changed-since` will only run over files directly changed. Often,
   test
 ```
 
+:::note Hint: Pants does not understand transitive third-party dependencies in this context.
+All of the `--changed-*` filters apply only to first party dependencies in your codebase.
```

Comment:
Maybe we can rephrase it a bit?

```suggestion
Changes to third-party dependencies (particularly, dependencies of dependencies) may not be surfaced as you expect via `--changed-*`. In particular, any change to a single dependency will consider all users of _any_ dependency changed, transitively. See [#12733](https://github.com/pantsbuild/pants/issues/12733)
```

I'm not 100% sure that's right though.

Reply:
Thank you, Huon! I've adopted your suggestion, that was helpful!

Reply:
Looks good, thanks.